homeassistant/components/mysensors/light.py

```diff
@@ -151,7 +150,7 @@ def _turn_on_rgb_and_w(self, hex_template: str, **kwargs: Any) -> None:
         if hex_template == "%02x%02x%02x%02x":
             if new_white is not None:
                 rgb.append(new_white)
-            else:
+            elif white is not None:
                 rgb.append(white)
```

Comment:
The rgb list needs to have the right lenght, maybe like this:

```suggestion
            elif white is not None:
                rgb.append(white)
            else:
                rgb.append(0)
```

Reply:
Yep, that makes sense 👍